Add singleProcessOOMKill option to kubelet

jaeyung1001 opened the pull request:
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds support for the `singleProcessOOMKill` option in the kubelet configuration via the kubelet_single_process_oom_kill variable. This option was introduced in Kubernetes v1.32 and, when set to `true`, causes OOM kills to be applied to individual processes rather than to the whole container cgroup.

- more detail on [link](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration)

**Which issue(s) this PR fixes**:
Fixes #12362

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Add support for the singleProcessOOMKill kubelet option (Kubernetes v1.32+) via the kubelet_single_process_oom_kill variable.
```

tico88612 commented:
The release note should not `NONE`, this is new feature for users, please edit the release note.

jaeyung1001 commented on docs/ansible/vars.md:
```diff
@@ -246,6 +246,8 @@ Stack](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/advanced/dn
 * *kubelet_max_parallel_image_pulls* - Sets the maximum number of image pulls in parallel. The value is `1` by default which means the default is serial image pulling, set it to a integer great than `1` to enable image pulling in parallel.
 
+
+
```
Oh, I just removed extra spaces. thank you!
- https://github.com/kubernetes-sigs/kubespray/pull/12365/commits/e477fb56095e1130b0f9df85a400cae52bb832b1

jaeyung1001 commented on roles/kubernetes/node/templates/kubelet-config.v1beta1.yaml.j2:
```diff
@@ -134,3 +134,4 @@ tracing:
   samplingRatePerMillion: {{ kubelet_tracing_sampling_rate_per_million }}
 {% endif %}
 maxParallelImagePulls: {{ kubelet_max_parallel_image_pulls }}
+singleProcessOOMKill: {{ kubelet_single_process_oom_kill }}
```
I added `kube_version` condition like below. Is it fine?
- https://github.com/kubernetes-sigs/kubespray/pull/12365/commits/0bbd75c210d00b963b67b9e242ab0d74e1e1a274

jaeyung1001 commented:
> The release note should not `NONE`, this is new feature for users, please edit the release note.

Thank you for advise. I updated release-note on description.

tico88612 commented:
/ok-to-test

jaeyung1001 commented:
singleProcessOOMKill The corresponding value does not have to be specified, so I annotated it and modified it to be used only when defined.

jaeyung1001 commented:
@tico88612 It's looks pass all CI test. Do you have any additional feedback?

jaeyung1001 commented:
> > 3. changed the code that not used `is defined`
> 
> What I wrote is not very precise, I actually want each value to be preset, not annotated.

Thank you for detail explaination. I have updated the value with detail description.

jaeyung1001 commented:
@tico88612 Hi :). Do you have any additional feedback?

VannTen commented:
Before merging this, I'd like to make the same point as in https://github.com/kubernetes-sigs/kubespray/issues/12337#issuecomment-3012975714 
TL;DR: What does this add that can't be accomplished with `kubelet_config_extra_args` ?
